File: PJDesignBackStageAPI/App.Model/Administrator/CreateOrUpdateGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App.Model
{
    public class CreateOrUpdateGroupRequest
    {
        public int? Id { get; set; }
        public string Name { get; set; } = null!;
        public List<UnitRight>? UnitRights { get; set; }
    }

    public class UnitRight
    {
        public int UnitId { get; set; }
        public int RightId { get; set; }
    }
}

File: PJDesignBackStageClient/src/app/features/administrator/feature-shared/models/create-or-update-group.ts
export class CreateOrUpdateGroupRequest {
  id: number;
  name: string;
  unitRights: UnitRight[] = [];

  constructor(id: number, name: string) {
    this.id = id;
    this.name = name;
  }
}

export class UnitRight {
  unitId: number;
  rightId: number;

  constructor(unitId: number, rightId: number) {
    this.unitId = unitId;
    this.rightId = rightId;
  }
}
